cosmos/operators/watcher.py

```diff
@@ -123,6 +124,17 @@ def _handle_startup_event(self, ev: EventMsg, startup_events: list[dict[str, Any
         ts_val = raw_ts.ToJsonString() if hasattr(raw_ts, "ToJsonString") else str(raw_ts)  # type: ignore[union-attr]
         startup_events.append({"name": info.name, "msg": info.msg, "ts": ts_val})
 
+    def _extract_compiled_sql_for_node_event(self, ev: EventMsg) -> str | None:
+        if getattr(ev.data.node_info, "resource_type", None) != "model":
+            return None
+        uid = ev.data.node_info.unique_id
+        node_path = str(ev.data.node_info.node_path)
+        package = uid.split(".")[1]
+        compiled_sql_path = Path.cwd() / "target" / "compiled" / package / "models" / node_path
+        if compiled_sql_path.exists():
+            return compiled_sql_path.read_text(encoding="utf-8").strip() or None
```

Comment:
It may be worth logging this scenario; it would be great if we could, in future, add telemetry, since it should not happen.